ocs_ci/utility/utils.py

```diff
@@ -692,22 +690,30 @@ def get_openshift_client(
             (default: config.RUN['client_version'])
         bin_dir (str): Path to bin directory (default: config.RUN['bin_dir'])
         force_download (bool): Force client download even if already present
+        skip_comparison (bool): Skip the comparison between the existing OCP client
+            version and the configured one.
 
     Returns:
         str: Path to the client binary
 
     """
     version = version or config.RUN["client_version"]
-    version = expose_ocp_version(version)
     bin_dir = os.path.expanduser(bin_dir or config.RUN["bin_dir"])
     client_binary_path = os.path.join(bin_dir, "oc")
     kubectl_binary_path = os.path.join(bin_dir, "kubectl")
     download_client = True
     client_version = None
+    try:
+        version = expose_ocp_version(version)
+    except Exception:
+        log.exception("Unable to expose OCP version, skipping client download.")
```

Comment:
```suggestion
    except Exception as ex:
        log.exception(f"Unable to expose OCP version, skipping client download. Exception: {ex}")
```

Please log the exception. If we do the general exception here, I would like to see it in the log as I pointed out in my previous comment.

Reply:
The [`exception()` method](https://docs.python.org/3/library/logging.html#logging.Logger.exception) will automatically include a traceback and the error in the logs, there is no need to explicitly push the exception into the message again:

>     Logs a message with level ERROR on this logger. The arguments are interpreted as for debug(). Exception info is added to the logging message. This method should only be called from an exception handler.


```
In [1]: import logging                                                                                                                  

In [2]: try: 
   ...:     print(1/0) 
   ...: except: 
   ...:     logging.exception("ops") 
   ...:                                                                                                                                 
ERROR:root:ops
Traceback (most recent call last):
  File "<ipython-input-2-2ba5b913f0db>", line 2, in <module>
    print(1/0)
ZeroDivisionError: division by zero

In [3]:
```

Reply:
Ah, I didn't realized and missed it's log.exception. I thought it was warning level in some of past comment hence I wanted to have this logged out. Sorry for confusion.